google/cloud/aiplatform/jobs.py

```diff
@@ -1913,3 +1918,522 @@ def run(
+                Dict[str, model_monitoring.objective.EndpointObjectiveConfig]):
+                A single config if it applies to all models, or a dictionary of
+                model_id: model_monitoring.objective.EndpointObjectiveConfig if
+                different model IDs have different configs.
+
+            deployed_model_ids (List[str]):
+                Optional. Use this argument to specify which deployed models to
+                apply the updated objective config to. If left unspecified, the same config
+                will be applied to all deployed models.
+
+        Raises:
+            RuntimeError, when the job isn't ready to be updated yet
+
+        """
+        current_job = self.api_client.get_model_deployment_monitoring_job(
+            name=self._gca_resource.name
+        )
+        update_mask: List[str] = []
+        if display_name:
+            update_mask.append("display_name")
+            current_job.display_name = display_name
+        if schedule_config:
+            update_mask.append("model_deployment_monitoring_schedule_config")
+            current_job.model_deployment_monitoring_schedule_config = schedule_config
+        if alert_config:
+            update_mask.append("model_monitoring_alert_config")
+            current_job.model_monitoring_alert_config = alert_config
+        if logging_sampling_strategy:
+            update_mask.append("logging_sampling_strategy")
+            current_job.logging_sampling_strategy = logging_sampling_strategy
+        if labels:
+            update_mask.append("labels")
+            current_job.lables = labels
+        if bigquery_tables_log_ttl:
+            update_mask.append("log_ttl")
+            current_job.log_ttl = bigquery_tables_log_ttl
+        if enable_monitoring_pipeline_logs:
+            update_mask.append("enable_monitoring_pipeline_logs")
+            current_job.enable_monitoring_pipeline_logs = (
+                enable_monitoring_pipeline_logs
+            )
+        if objective_configs:
+            update_mask.append("model_deployment_monitoring_objective_configs")
+            current_job.model_deployment_monitoring_objective_configs = (
+                ModelDeploymentMonitoringJob._parse_configs(
+                    objective_configs,
+                    current_job.endpoint,
+                    deployed_model_ids,
+                )
+            )
+        if self.state == gca_job_state.JobState.JOB_STATE_RUNNING:
+            self.api_client.update_model_deployment_monitoring_job(
+                model_deployment_monitoring_job=current_job,
+                update_mask=field_mask_pb2.FieldMask(paths=update_mask),
+            )
+        else:
+            raise RuntimeError(
+                "The monitoring job can only be updated under running state, the current state is: %s"
+                % self.state
+            )
+
+    def pause(self) -> "ModelDeploymentMonitoringJob":
```

Comment:
This doesn't return self.

Reply:
fixed. i also made `update` return self as well in order to allow for method cascading